src/containers/List/ListFilters/ListFilters.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from 'react'
+import { Dropdown } from 'semantic-ui-react'
+import { useRouter } from '../../../utils/hooks/useRouter'
+import { FilterDefinitions } from '../../../utils/types'
+import BooleanFilter from './BooleanFilter'
+import { EnumFilter, SearchableListFilter, StaticListFilter } from './OptionFilters'
+import { GetMethodsForOptionFilter } from './types'
+
+const getArrayFromString = (string: string = '') =>
+  string.split(',').filter((option) => !!option.trim())
+
+const getDisplayableFilters = (filterDefinitions: FilterDefinitions) => {
+  const result: FilterDefinitions = {}
+  Object.entries(filterDefinitions).forEach(([filterName, filterValue]) => {
+    // Missing title determines if filters should be shown in UI or not
+    if (!filterValue.title) return
+    result[filterName] = filterValue
+  })
+
+  return result
+}
+
+const ListFilters: React.FC<{ filterDefinitions: FilterDefinitions; filterListParameters: any }> =
+  ({ filterDefinitions, filterListParameters }) => {
+    const { query, updateQuery } = useRouter()
+
+    const displayableFilters = getDisplayableFilters(filterDefinitions)
+
+    const filterNames = Object.keys(displayableFilters)
+    const [activeFilters, setActiveFilters] = useState<string[]>([])
+
+    // Add fitlers from URL to activeFilters, filters may not have criteria/options yet thus we have to use activeFilter state variable
+    useEffect(() => {
+      const filersInQuery = filterNames.filter((filterName) => !!query[filterName])
+      const newFilters = filersInQuery.filter((filterName) => !activeFilters.includes(filterName))
+      setActiveFilters([...activeFilters, ...newFilters])
+    }, [query])
+
+    // Filter criteria/options states is provided by query URL, methods below are get and set query fiter criteria
+    const getMethodsForOptionFilter: GetMethodsForOptionFilter = (filterName) => ({
+      getActiveOptions: () => getArrayFromString(query[filterName]),
+      setActiveOption: (option: string) =>
+        updateQuery({ [filterName]: [...getArrayFromString(query[filterName]), option].join(',') }),
+      setInactiveOption: (option: string) =>
+        updateQuery({
+          [filterName]: getArrayFromString(query[filterName])
+            .filter((_option) => _option != option)
+            .join(','),
+        }),
+    })
+
+    const addFilter = (filterName: string) => setActiveFilters([...activeFilters, filterName])
+    const getOnRemove = (filterName: string) => () => {
+      setActiveFilters(activeFilters.filter((_filterName) => _filterName !== filterName))
+      updateQuery({
+        [filterName]: undefined,
+      })
+    }
+
+    const availableFilterNames = filterNames.filter(
+      (filterName) => !activeFilters.includes(filterName)
+    )
+
+    return (
+      <div id="list-filter">
+        <Dropdown text="+ Filter" icon={null}>
```

Comment:
Localise "+ Filter" string?

Reply:
Also -- should the "+" be an icon? (You can maybe re-use the styling for the "+ New Application" button on the right)

Reply:
76c191b7e88206665d7a00966598a805b761acaf